### ISSUE TITLE
Add a way to detect if callback has been called with error because images have changes

### ISSUE THREAD

**Issue by remorses** [jihchi/dify-bin#5]
Currently the callback will be called with an error if changes are detected, i don't know how to verify if the error has been caused by detected changes or another unexpected error

Maybe i can check for a special exit code on the error to see if the command failed because changes have been detected?

It would be cool to add a `error.changesDetected = true` when the command fails because of images being different

**Comment by jihchi** [comment]
I'm reopening this issue even though we now have documented a work-around in README (#6, thanks @remorses ), I think we should have a better way to handle this case.

**Comment by jihchi** [comment]
I'm thinking to move the logic from the README as an internal logic and provide an abstraction on it, for example it may look like this:

```js
dify(['--output', 'diff.png', 'left.jpg', 'right.jpg'], (err, out) => {
  if (err) {
    console.error(err.stderr);
    process.exit(err.code);
    return;
  }
  if (out.different) {
    console.log('Changes detected');
  } else {
    console.log('No changes detected');
  }
});
```

@remorses does this fulfill your use case?

**Comment by remorses** [comment]
Yes, this way it's cleaner

I would use `out.isDifferent` instead of `out.different`

**Comment by jihchi** [comment]
I've published [v0.0.10](https://github.com/jihchi/dify-bin/releases/tag/v0.0.10)!
